driver-core/src/main/com/mongodb/client/model/expressions/StringExpression.java

```diff
@@ -25,34 +25,142 @@
+     */
     IntegerExpression strLenBytes();
 
+    /**
+     * The substring of {@code this} string, from the {@code start} index
+     * inclusive, and including the specified {@code length}, up to
+     * the end of the string.
+     *
+     * <p>Warning: the index position is in UTF-8 code points, not in
+     * UTF-8 encoded bytes.
+     *
+     * @param start the start index in UTF-8 code points.
+     * @param length the length in UTF-8 code points.
+     * @return the resulting value.
+     */
     StringExpression substr(IntegerExpression start, IntegerExpression length);
 
+    /**
+     * The substring of {@code this} string, from the {@code start} index
+     * inclusive, and including the specified {@code length}, up to
+     * the end of the string.
+     *
+     * <p>Warning: the index position is in UTF-8 code points, not in
+     * UTF-8 encoded bytes.
+     *
+     * @param start the start index in UTF-8 code points.
+     * @param length the length in UTF-8 code points.
+     * @return the resulting value.
+     */
     default StringExpression substr(final int start, final int length) {
         return this.substr(of(start), of(length));
     }
 
+    /**
+     * The substring of {@code this} string, from the {@code start} index
+     * inclusive, and including the specified {@code length}, up to
+     * the end of the string.
+     *
+     * <p>The index position is in UTF-8 encoded bytes, not in
+     * UTF-8 code points.
+     *
+     * @param start the start index in UTF-8 encoded bytes.
+     * @param length the length in UTF-8 encoded bytes.
+     * @return the resulting value.
+     */
     StringExpression substrBytes(IntegerExpression start, IntegerExpression length);
 
+    /**
+     * The substring of {@code this} string, from the {@code start} index
+     * inclusive, and including the specified {@code length}, up to
+     * the end of the string.
+     *
+     * <p>The index position is in UTF-8 encoded bytes, not in
+     * UTF-8 code points.
+     *
+     * @param start the start index in UTF-8 encoded bytes.
+     * @param length the length in UTF-8 encoded bytes.
+     * @return the resulting value.
+     */
     default StringExpression substrBytes(final int start, final int length) {
         return this.substrBytes(of(start), of(length));
     }
 
+    /**
+     * Converts {@code this} string to an {@linkplain IntegerExpression integer}.
+     *
+     * <p>This will cause an error if this string does not represent an integer.
+     *
+     * @return the resulting value.
+     */
     IntegerExpression parseInteger();
 
+    /**
+     * Converts {@code this} string to a {@linkplain DateExpression date}.
+     *
+     * <p>This will cause an error if this string does not represent a valid
+     * date string (such as "2018-03-20", "2018-03-20T12:00:00Z", or
+     * "2018-03-20T12:00:00+0500").
+     *
+     * @return the resulting value.
+     */
     DateExpression parseDate();
```

Comment:
I seems that we should specify that this method behaves just like `parseDate(StringExpression format)` with the default format, which we should also specify.